src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -1166,17 +1169,18 @@ size_t ObjectSynchronizer::in_use_list_ceiling() {
 
 void ObjectSynchronizer::dec_in_use_list_ceiling() {
   Atomic::add(&_in_use_list_ceiling, (jint)-AvgMonitorsPerThreadEstimate);
-#ifdef ASSERT
-  size_t l_in_use_list_ceiling = in_use_list_ceiling();
-#endif
-  assert(l_in_use_list_ceiling > 0, "in_use_list_ceiling=" SIZE_FORMAT
-         ": must be > 0", l_in_use_list_ceiling);
```

Comment:
Did you find that this can go negative?  I can see that it could go to zero at shutdown maybe.

Reply:
In a previous round I changed the assert to:
```assert(l_in_use_list_ceiling >= 0, ...```
and the Linux build complained about the assert always being true.
This is because size_t is unsigned. (No complaint on macOSX.)

Reply:
> what would happen if this option was set to max_jint? Is it a reasonable max?

The _in_use_list_ceiling would be set to a very large number and we would
be allowing a very large number of monitors before a deflation cycle, i.e., we
would probably never async deflate.

Reply:
Oh, Ok, on both points.